src/HttpOverStream.Server.Owin/CustomListenerHost.cs

```diff
@@ -117,25 +131,31 @@ public static IDisposable Start(Action<IAppBuilder> startup, IListen listener)
 
         private static IDisposable Start(StartContext context, IListen listener)
         {
-            context.ServerFactory = new ServerFactoryAdapter(new CustomListenerHostFactory(listener));
             IServiceProvider services = ServicesFactory.Create();
             var engine = services.GetService<IHostingEngine>();
+            context.ServerFactory = new ServerFactoryAdapter(new CustomListenerHostFactory(listener));            
             return engine.Start(context);
         }
 
         private class CustomListenerHostFactory
         {
             private readonly IListen _listener;
+            private IAppBuilder _builder;
 
             public CustomListenerHostFactory(IListen listener)
             {
                 _listener = listener;
             }
             public IDisposable Create(Func<IDictionary<string, object>, Task> app, IDictionary<string, object> properties)
             {
-                var result = new CustomListenerHost(_listener, app);
-                result.Start();
-                return result;
+                var host = new CustomListenerHost(_listener, app, _builder);
+                host.Start();
+                return host;
+            }
+
+            public void Initialize(IAppBuilder builder)

```

Comment:
Is this function called somewhere?

Reply:
It is called by the Owin infrastructure yes.
The ServerFactoryAdapter thing is really strange as it does not rely on an interface but using reflection to discover methods... (as you can guess, without access to the source code of Owin, implementing this would have been... difficult)